### ISSUE TITLE
Password with symbols can cause errors parsing rally.ini file

### ISSUE THREAD

**Issue by cdahlqvist** [elastic/rally#519]
**Rally version**: 0.11.0

**OS version**: Ubuntu 16.04

**Description of the problem including expected versus actual behavior**:
I specified an Elastic Cloud cluster as the metrics store and picked a password that contained a `$` followed by a character. This resulted in the following error when I tried to run the eventdata track:

```
Traceback (most recent call last):
  File "/usr/local/bin/esrally", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/esrally/rally.py", line 588, in main
    ensure_configuration_present(cfg, args, sub_command)
  File "/usr/local/lib/python3.6/site-packages/esrally/rally.py", line 428, in ensure_configuration_present
    cfg.load_config(auto_upgrade=True)
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 206, in load_config
    self.migrate_config()
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 239, in migrate_config
    migrate(self.config_file, self._stored_config_version(), Config.CURRENT_CONFIG_VERSION)
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 588, in migrate
    .format(config_file.location, PROGRAM_NAME))
esrally.config.ConfigError: ('The config file in /home/ubuntu/.rally/rally.ini is too old. Please delete it and reconfigure Rally from scratch with esrally configure.', None)
```

**Comment by danielmitterdorfer** [comment]
It looks as if this problem is caused by something else. The error message says:

> The config file in /home/ubuntu/.rally/rally.ini is too old. Please delete it and reconfigure Rally from scratch with esrally configure.

We recently dropped support for ancient config file format versions (i.e. `config.version` < 12) and this is why it fails. I suggest you reconfigure Rally with `esrally configure --advanced-config` and retry.

**Comment by cdahlqvist** [comment]
Sorry. Must have copied the wrong stack trace. I now configured it locally using the `configure` utility and got the following:

```
esrally
Traceback (most recent call last):
  File "/usr/local/bin/esrally", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/esrally/rally.py", line 580, in main
    arg_parser = create_arg_parser()
  File "/usr/local/lib/python3.6/site-packages/esrally/rally.py", line 107, in create_arg_parser
    cfg.load_config()
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 204, in load_config
    self._do_load_config()
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 214, in _do_load_config
    self._fill_from_config_file(config)
  File "/usr/local/lib/python3.6/site-packages/esrally/config.py", line 233, in _fill_from_config_file
    self.add(Scope.application, section, key, config[section][key])
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/configparser.py", line 1234, in __getitem__
    return self._parser.get(self._name, key)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/configparser.py", line 800, in get
    d)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/configparser.py", line 455, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/configparser.py", line 518, in _interpolate_some
    "found: %r" % (rest,))
configparser.InterpolationSyntaxError: '$' must be followed by '$' or '{', found: '$tPa$$word'
```

**Comment by danielmitterdorfer** [comment]
Thanks for reporting. Now I could reproduce it. Fixed in #524 and scheduled to be released with 1.0.0.
